warehouse/oidc/models/_core.py

```diff
@@ -102,6 +102,39 @@ def check_existing_jti(
     return True
 
 
+def verify_url_from_reference(*, reference_url: str, url: str) -> bool:
```

Comment:
Nitpick: I think ideally this would live somewhere else, since it doesn't have much to do with the core OIDC models. But I don't have any good ideas about where it should live...

Reply:
Agreed. At the very least, let's move it to `warehouse/oidc/utils.py`.

Reply:
Moving it to `utils` led to a circular import but I created a new `urls.py` file for the function.